### DISCUSSION TITLE
How to use pagination?

### DISCUSSION THREAD

**Discussion by brianwx5** [sensedeep/dynamodb-onetable#335] [Q&A] [answered]
I tried using pagination as it said in the document for v1.8.1.

`let firstPage = await User.find({accountId}, {limit})`
`let secondPage = await User.find({accountId}, {limit, next: firstPage.next})`
`let previousPage = await User.find({accountId}, {limit, prev: secondPage.prev})`

When I get to the second page again using the previousPage.next, it skips to the next item. Wasn't sure if I'm using it correctly or not.
Example of it would be:
- Page 1: [1, 2, 3]
- Page 2: [4, 5, 6]
- Back to Page 1: [1, 2, 3]
- Back to Page 2: [5, 6, 7] but expected [4, 5, 6]

There's other scenarios weird paging scenarios but this was the one I saw. I'm not sure if this was fixed in a later release.

**Comment by mobsense** [comment, marked as answer]
If you have a test case, we can move this over to issues and we'll try to see what is going on.

**Reply by dev-embedthis** [reply]
Some work was done in 2.6.x with pagination and may have fixed your issue.
